### PR TITLE
Add devcontainer/Codespaces setup so beginners can run examples without worrying about setup

### PR DESCRIPTION
**Pull request by jonathanagustin**
Adding a devcontainer/Codespaces setup will allow beginners to immediately use OpenAI's API without worrying about setting up their environment. The setup includes cloning the `openai-cookbook` repository and installing most of the dependencies required to run the examples. After building and setting the API key, a user can immediately start executing the examples.

With greater accessibility of infrastructure, more people can learn and use OpenAI's Python API. If this PR is merged, the README.md should include notice to users about potential Codespace development on the cloud without worrying about the environment.

I chose to scope the entirety of the PR to the `.devcontainer` folder to contain the entirety of the feature. I chose a Python3.8 image because some dependencies required it.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+    "image": "mcr.microsoft.com/devcontainers/python:0-3.8",
```
[the remark this answers is not in the document]
Comment (X11tpro) [reply .devcontainer/devcontainer.json:2]: Chosen phyton 3.7, can i say this is the new one that will emerge .

[the remark this answers is not in the document]
Comment (X11tpro) [reply .devcontainer/devcontainer.json:2]: Corrections phyton 3.8

**Comment by rattrayalex** [comment]
This has been added! Thanks for the PR!

cc @RobertCraigie

**Comment by rattrayalex** [comment]
@jonathanagustin please try what we have on the main branch and let us know what you think!

**Comment by jonathanagustin** [comment]
@rattrayalex 

Looks good - I'm glad this was able to get in one way or another!

**Comment by goldenk4** [comment]
I will test it out and keep you updated

**Pull request closed without merging**